traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm

```diff
@@ -2601,16 +2601,14 @@ sub parent_dot_config { #fix qstring - should be ignore for quika
 
 		my $default_dest_text;
 		my $parent_select_alg = $self->profile_param_value( $server_obj->profile->id, 'parent.config', 'algorithm', undef );
+		$default_dest_text .= "dest_domain=. ";
 		if ( defined($parent_select_alg) && $parent_select_alg eq 'consistent_hash' ) {
-			$default_dest_text .= "dest_domain=. ";
 			$default_dest_text .= $parents . $secparents;
-			$default_dest_text .= " round_robin=consistent_hash go_direct=false";
 		}
 		else {    # default to old situation.
-			$default_dest_text .= "dest_domain=. ";
 			$default_dest_text .= $parents;
-			$default_dest_text .= " round_robin=urlhash go_direct=false";
 		}
+		$default_dest_text .= " round_robin=consistent_hash go_direct=false";
```

Comment:
Couldn't tell you. The original code appears to ignore the value of the parent selection algorithm entirely. It was just hard-coded to use `consistent_hash` whenever the algorithm was defined and the erroneous `urlhash` in all other cases.

Reply:
Understood.  I know that Cisco is using 'latched'.  Not sure how they configured that with this code though.

Reply:
I have not, and I'll tell you why:

> _"Not sure how they configured that with this code"_

because I'm not a Perl wizard and I have no idea how that was working in the first place, so I left the logic as intact as possible. I'd certainly agree that the proper behavior would apparently be to set the algorithm as you suggested, but I think that falls outside the scope of this PR which is only meant to fix #3071 .